plugins/woocommerce/includes/class-wc-shortcodes.php

```diff
@@ -575,6 +572,15 @@ public static function product_page( $atts ) {
 
 		$single_product = new WP_Query( $args );
 
+		if (
+			! isset( $force_rendering ) &&
+			$single_product->have_posts() &&
```

Comment:
Shouldn't we be able to leave early if this is false? Right after the query we could add ...

```php
		if ( ! $single_product->have_posts() ) {
			return '';
		}
```

What do you think?

Reply:
I think you might be right but I didn't want to add another condition & early return in to this PR since this wasn't part of this method prior to https://github.com/Automattic/woocommerce/pull/298 anyway.

What do you think about creating this in a separate PR as a follow up? If you think it definitely should be included here then I can go ahead and add it.

Reply:
It will not render anything anyway since this check is also done here I believe https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-shortcodes.php#L628-L635

Reply:
Yep, additional PR sounds good. Thanks!